### PR TITLE
[m] Upgrade hibernate to 5.2.17

### PR DESCRIPTION
**Pull request by vritant**

### CODE REVIEW

**Comment by vritant** [comment]
@candlepin-pull-request-bot test this please

File: server/src/test/java/org/candlepin/test/SessionWrapper.java [hunk 1]
```diff
@@ -16,65 +16,83 @@
...
+import javax.persistence.criteria.Selection;
+import javax.persistence.metamodel.Metamodel;
 
 /**
  * The SessionWrapper is a utility class intended to be used in places where we need to spy on an
  * existing session instance, but are unable to due to the "final" nature of Hibernate's
  * SessionImpl class.
  */
-public class SessionWrapper implements Session, SessionImplementor {
+public class SessionWrapper implements SessionImplementor {
```
Comment (wottop) [inline server/src/test/java/org/candlepin/test/SessionWrapper.java]: If you extend AbstractSessionImpl, wouldn't it reduce the number of no-op implementations you added?

Comment (vritant) [reply server/src/test/java/org/candlepin/test/SessionWrapper.java]: perhaps. reduces the class SessionWrapper from a 1600 line class to 900 line class...but we still have a *lot* of boilerplate code.
but that class implicitly introduces more interfaces than needed, so Imo we should just stick with this.
thoughts?

Comment (wottop) [reply server/src/test/java/org/candlepin/test/SessionWrapper.java]: You convinced me. ack.

**Pull request merged**